### ISSUE TITLE
Cannot build Version 4.7.13 failing with errors.

### ISSUE THREAD

**Issue by alwendt4** [tinymce/tinymce#4454]
Steps to reproduce the problem (Windows Server 2016) ...

- Go to https://github.com/tinymce/tinymce.  Click Releases link.  Download and unzip verrsion 4.7.13 (file tinymce-4.7.13.zip).  Open the readme.md file and continue following steps in the readme.md ...
- Open Git Bash console.
- cd tinymce/
- $ npm i -g grunt-cli
+ grunt-cli@1.2.0
updated 1 package in 3.416s
- $ npm install
npm WARN deprecated coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)

> uglifyjs-webpack-plugin@0.4.6 postinstall C:\users\Administrator\Desktop\test\tinymce\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js

npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN ajv-keywords@3.2.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 923 packages from 723 contributors and audited 11006 packages in 77.537s
found 7 vulnerabilities (3 low, 4 moderate)
  run `npm audit fix` to fix them, or `npm audit` for details
- $ grunt
$ grunt
Running "validateVersion" task

Running "shell:tsc" (shell) task
src/core/main/ts/keyboard/CefNavigation.ts(231,31): error TS1005: ')' expected.
src/core/main/ts/keyboard/CefNavigation.ts(231,40): error TS1005: ',' expected.
src/core/main/ts/keyboard/CefNavigation.ts(231,41): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/CefNavigation.ts(231,43): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/CefNavigation.ts(244,28): error TS1005: ')' expected.
src/core/main/ts/keyboard/CefNavigation.ts(244,37): error TS1005: ',' expected.
src/core/main/ts/keyboard/CefNavigation.ts(244,38): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/CefNavigation.ts(244,40): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/TableNavigation.ts(150,31): error TS1005: ')' expected.
src/core/main/ts/keyboard/TableNavigation.ts(150,40): error TS1005: ',' expected.
src/core/main/ts/keyboard/TableNavigation.ts(150,41): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/TableNavigation.ts(150,43): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/TableNavigation.ts(160,31): error TS1005: ')' expected.
src/core/main/ts/keyboard/TableNavigation.ts(160,40): error TS1005: ',' expected.
src/core/main/ts/keyboard/TableNavigation.ts(160,41): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/TableNavigation.ts(160,43): error TS1134: Variable declaration expected.
Warning: Command failed: node ./node_modules/typescript/bin/tsc
 Use --force to continue.

Aborted due to warnings.

Can someone PLEASE tell me what I have to do to build this?

**Comment by Georg-Git** [comment]
#4441 
 "..just delete the node_modules//node_modules folder and it should build fine.."

**Comment by Georg-Git** [comment]
Fresh installation of node,js including npm.

file tinymce-4.7.13.zip => same errors in tsc

In the last days there are lot of security updates in the modules tinyMCE depends on.

**Comment by BradMcGonigle** [comment]
I'm using [`nvm`](https://github.com/creationix/nvm) with a fresh install of node v10.5.0 and npm v6.1.0 and still running into this issue when trying running `grunt`.

I've also double checked and there is no extra `node_modules` folder in `node_modules/@ephox/alloy`.

```
Running "validateVersion" task

Running "shell:tsc" (shell) task
src/core/main/ts/keyboard/CefNavigation.ts(231,31): error TS1005: ')' expected.
src/core/main/ts/keyboard/CefNavigation.ts(231,40): error TS1005: ',' expected.
src/core/main/ts/keyboard/CefNavigation.ts(231,41): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/CefNavigation.ts(231,43): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/CefNavigation.ts(244,28): error TS1005: ')' expected.
src/core/main/ts/keyboard/CefNavigation.ts(244,37): error TS1005: ',' expected.
src/core/main/ts/keyboard/CefNavigation.ts(244,38): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/CefNavigation.ts(244,40): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/TableNavigation.ts(150,31): error TS1005: ')' expected.
src/core/main/ts/keyboard/TableNavigation.ts(150,40): error TS1005: ',' expected.
src/core/main/ts/keyboard/TableNavigation.ts(150,41): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/TableNavigation.ts(150,43): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/TableNavigation.ts(160,31): error TS1005: ')' expected.
src/core/main/ts/keyboard/TableNavigation.ts(160,40): error TS1005: ',' expected.
src/core/main/ts/keyboard/TableNavigation.ts(160,41): error TS1134: Variable declaration expected.
src/core/main/ts/keyboard/TableNavigation.ts(160,43): error TS1134: Variable declaration expected.
Warning: Command failed: node ./node_modules/typescript/bin/tsc
 Use --force to continue.
Error: Command failed: node ./node_modules/typescript/bin/tsc

    at ChildProcess.exithandler (child_process.js:291:12)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:961:16)
    at Socket.stream.socket.on (internal/child_process.js:380:11)
    at Socket.emit (events.js:182:13)
    at Pipe._handle.close (net.js:595:12)


Aborted due to warnings.
```

**Comment by Georg-Git** [comment]
I suppose building tiny with the latest existing tiny releases will not be possible anymore!

In the file package.json a lot of dependencies of tiny are marked with version "latest" or with minimum versions.

In May npm introduced automatic security audits and the users were shocked by "vulnerabilities" warnings during the building process.
This warnings lead to hectic activities and updates in a lot of npm modules.
And obviously they had become incompatible.

During the last years we had seen a release update of tiny twice a month - but now the latest tiny release is over 6 weeks old ;-)

But nevertheless - building tiny with master code is still possible.

**Comment by Afraithe** [comment]
Reason for less updates lately is that we are working on some larger tasks. And be sure we are looking into these build issues.

**Comment by Georg-Git** [comment]
This issue is with zip file version 4.7.13.

You just published 4.8.0 - and building is now possible again..

But with version 4.8.0 a warning in the rollup task makes me nervous:

```
Running "rollup:core" (rollup) task
The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
```

**Comment by lnewson** [comment]
Closing this as 4.7.x is very old now and no longer supported. If you're having issues with 4.9.x or 5.x please do let us know though and we'll investigate.
